runtime_scan/pkg/provider/docker/client.go

```diff
@@ -0,0 +1,456 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/api/types/volume"
+	"github.com/docker/docker/client"
+	"gopkg.in/yaml.v2"
+
+	"github.com/openclarity/vmclarity/api/models"
+	"github.com/openclarity/vmclarity/runtime_scan/pkg/provider"
+	"github.com/openclarity/vmclarity/shared/pkg/families"
+	"github.com/openclarity/vmclarity/shared/pkg/log"
+)
+
+var (
+	// mountPointPath defines the location in the container where assets will be mounted.
+	mountPointPath = "/mnt/snapshot"
+	// helperImage defines helper container image that performs init tasks.
+	helperImage = "alpine"
```

Comment:
I don't think my comment was addressed about making this configurable and also pinning a version right now we could get any alpine, even one with a nasty malware if someone poisoned upstream

Reply:
I rather helper image was configurable through an ENV with a default, if for whatever reason dockerhub isn't accessible or a deployer must use a local docker registry because they aren't allowed internet access. We should allow it to be overridden.

Reply:
Sure, can't see why not. I summarised it in a new review item @sambetts-cisco